admin_assignment_update.php

```diff
@@ -1,95 +1,45 @@
-		 $file = $_FILES['moduleassignment_7']['tmp_name'];
-		 move_uploaded_file($file,$destination);
-	
-		 $moduleassignment_8 =$_FILES['moduleassignment_8']['name'];
-		 $destination = 'moduleassignmentsforStudents/' . $moduleassignment_8 ;
-		 $extention = pathinfo($moduleassignment_8, PATHINFO_EXTENSION);
-		 $file = $_FILES['moduleassignment_8']['tmp_name'];
-		 move_uploaded_file($file,$destination);
-	
-		 $moduleassignment_9 =$_FILES['moduleassignment_9']['name'];
-		 $destination = 'moduleassignmentsforStudents/' . $moduleassignment_9 ;
-		 $extention = pathinfo($moduleassignment_9, PATHINFO_EXTENSION);
-		 $file = $_FILES['moduleassignment_9']['tmp_name'];
-		 move_uploaded_file($file,$destination);
-	
-	     $moduleassignment_10 =$_FILES['moduleassignment_10']['name'];
-		 $destination = 'moduleassignmentsforStudents/' . $moduleassignment_10 ;
-		 $extention = pathinfo($moduleassignment_10, PATHINFO_EXTENSION);
-		 $file = $_FILES['moduleassignment_10']['tmp_name'];
-		 move_uploaded_file($file,$destination);
-   
-   $query = "UPDATE module_assignments SET  assignmnt_id='$assignmnt_id', batch_id='$batch_id', moduleassignment_1='$moduleassignment_1', moduleassignment_2='$moduleassignment_2', moduleassignment_3='$moduleassignment_3', moduleassignment_4='$moduleassignment_4' , moduleassignment_5='$moduleassignment_5', moduleassignment_6='$moduleassignment_6', moduleassignment_7='$moduleassignment_7', moduleassignment_8='$moduleassignment_8', moduleassignment_9='$moduleassignment_9', moduleassignment_10='$moduleassignment_10' WHERE assignmnt_id= $assignmnt_id " ;
-	
-	
-	
-	
-	
-	$query_run = mysqli_query($conn, $query);
+	$assignmnt_id = $_POST['assignmnt_id']; 
+	$batch_id = $_POST['batch_id'];
 
-	
-	if($query_run)
-	{
-		
-	echo "submitted";	
-   header('location:admin_assignments.php') ;     
-		
+	// Upload files
+	$moduleAssignments = [];
+	for ($i = 1; $i <= 10; $i++) {
+		if (!empty($_FILES["moduleassignment_$i"]["name"])) {
+			$filename = basename($_FILES["moduleassignment_$i"]["name"]);
+			$destination = 'moduleassignmentsforStudents/' . $filename;
+			move_uploaded_file($_FILES["moduleassignment_$i"]["tmp_name"], $destination);
+			$moduleAssignments[] = $filename;
+		}
 	}
-	else
-	{
+
+	// Build query
+	$query = "UPDATE module_assignments SET assignmnt_id=?, batch_id=?, ";
+	for ($i = 1; $i <= count($moduleAssignments); $i++) {
+		$query .= "moduleassignment_$i=?,";
+	}
+	$query = rtrim($query, ',') . " WHERE assignmnt_id=?";
+
+	// Execute query using prepared statement
+	$stmt = mysqli_prepare($conn, $query);
+	$params = array_merge([$assignmnt_id, $batch_id], $moduleAssignments, [$assignmnt_id]);
+	mysqli_stmt_bind_param($stmt, str_repeat('s', count($params)), ...$params);
+	$query_run = mysqli_stmt_execute($stmt);
+
+	// Handle result
+	if ($query_run) {
+		echo "submitted";    
+		header('location:admin_assignments.php');
+	} else {
 		echo "form not submitted";
+	}
 }
```

Comment:
This appears to be processing a form submission to update module assignments for students. While the code works, there are several improvements that can be made to optimize it.

Use prepared statements instead of concatenating variables into the SQL query to prevent SQL injection attacks.

Refactor the file upload code into a separate function to reduce duplication and increase readability.

Instead of uploading each file individually and hard-coding the file names, use an array to upload files dynamically based on the number of files selected by the user.